File: backend/app/schemas/ticket.py
from pydantic import BaseModel
from typing import Optional


class TicketBase(BaseModel):
    seat_id: int
    ticket_type_id: int
    status: str = "available"


class TicketCreate(TicketBase):
    pass


class TicketUpdate(BaseModel):
    owner_id: Optional[int] = None
    status: Optional[str] = None


class TicketResponse(TicketBase):
    ticket_id: int
    owner_id: Optional[int] = None

    class Config:
        from_attributes = True


class TicketTypeBase(BaseModel):
    event_id: int
    name: str
    price: float


class TicketTypeCreate(TicketTypeBase):
    pass


class TicketTypeResponse(TicketTypeBase):
    ticket_type_id: int

    class Config:
        from_attributes = True


class SectionTicketTypeMapping(BaseModel):
    section_id: int
    ticket_type_id: int


class BulkTicketGenerateRequest(BaseModel):
    section_id: int
    ticket_type_id: int


class BulkTicketGenerateResponse(BaseModel):
    tickets_created: int
    section_id: int
    ticket_type_id: int
    message: str


class EventBulkGenerateRequest(BaseModel):
    event_id: int


class EventBulkGenerateResponse(BaseModel):
    tickets_created: int
    event_id: int
    mappings_processed: int
    message: str

File: frontend/src/types/ticket.ts
export interface TicketType {
  ticket_type_id: number
  event_id: number
  name: string
  price: number
}

export interface TicketTypeCreate {
  event_id: number
  name: string
  price: number
}

export interface TicketBase {
  seat_id: number
  ticket_type_id: number
  status: TicketStatus
}

export interface TicketCreate extends TicketBase {}

export interface TicketUpdate {
  owner_id?: number
  status?: TicketStatus
}

export interface TicketResponse extends TicketBase {
  ticket_id: number
  owner_id?: number
}

export enum TicketStatus {
  AVAILABLE = 'available',
  RESERVED = 'reserved',
  SOLD = 'sold'
}

export interface MyTicket {
  ticket_id: number
  seat_id: number
  ticket_type_id: number
  status: string
  ticket_type_name: string
  price: number
  event_id: number
  event_name: string
  event_date: string
  venue_name: string
  venue_city: string
}

export interface CartItem {
  ticket: TicketResponse
  ticketType: TicketType
  eventId: number
  eventName: string
  eventDate: string
  venueName: string
}
